### DISCUSSION TITLE
Game Mode Switcher

### DISCUSSION THREAD

**Discussion by KyllianGamer** [Minestom/Minestom#1830] [Help] [answered]
So apparently there's a game mode switcher menu, didn't know this. However, when trying to use it on my Minestom server I get a client message "no permission". What permission is this exactly and how would I give it to the player?

**Comment by dev-hydrogen** [comment, marked as answer]
It is the operator level of the player, player.setPermissionLevel(int)
I think a value between 2-4 is required. <0 & >4 is an exception

**Reply by dev-hydrogen** [reply]
You also most likely need a /gamemode command that matches vanilla syntax

**Reply by emortaldev** [reply]
Yes you do need a gamemode command with the same syntax. I had the gamemode switcher working before.

**Reply by Tofaa2** [reply]
> Yes you do need a gamemode command with the same syntax. I had the gamemode switcher working before.

Community: we want data driven entities and itemstacks
Mojang: a gui of qol predefined commands is the best i can do, take it or leave it

**Reply by KyllianGamer** [reply]
> You also most likely need a /gamemode command that matches vanilla syntax

So I you could use a resource pack to make the menu look different and then listen for that command to handle each thing?

Actually, just remembered the gui requires the most absurd combination of keys to be opened, so nevermind.

Thanks for the answer though.
